src/resource_clients/dataset.ts

```diff
@@ -156,29 +123,75 @@ class DatasetClient extends ResourceClient {
         });
     }
 
-    /**
-     * @param response
-     * @return {PaginationList}
-     * @private
-     */
-    _createPaginationList(response) {
+    private _createPaginationList<D>(response: AxiosResponse<D[]>): PaginatedList<D> {
         return {
             items: response.data,
             total: Number(response.headers['x-apify-pagination-total']),
             offset: Number(response.headers['x-apify-pagination-offset']),
             count: response.data.length, // because x-apify-pagination-count returns invalid values when hidden/empty items are skipped
             limit: Number(response.headers['x-apify-pagination-limit']), // API returns 999999999999 when no limit is used
+            // TODO: will the API return this
+            desc: false,
         };
     }
 }
 
-module.exports = DatasetClient;
+type MaybeArray<T> = T | T[];
 
-/**
- * @typedef {object} PaginationList
- * @property {object[]} items - List of returned objects
- * @property {number} total - Total number of objects
- * @property {number} offset - Number of objects that were skipped
- * @property {number} count - Number of returned objects
- * @property {number} limit - Requested limit
- */
+export interface Dataset {
+    id: string;
+    name: string;
```

Comment:
name is definitely optional

Reply:
What docs are you using exactly? https://docs.apify.com/api/v2 have a reputation of being terribly outdated even though I pointed that out several times.

Reply:
E.g. https://api.apify.com/v2/datasets/8kUL85wh7Zhu9KkJx

Reply:
> What docs are you using exactly? [docs.apify.com/api/v2](https://docs.apify.com/api/v2) have a reputation of being terribly outdated even though I pointed that out several times.

That's the docs I'm using. 😱
Do you have any that might be closer-to-current-returns, since throwing random API calls isn't the best idea / ideal..

Reply:
Haha, this is what I got months ago when testing :) https://github.com/metalwarrior665/apify-client-rs/blob/master/src/datasets.rs#L139
But since we probably don't have a real source of truth, it will be a combination of:
- Finding some types in the SDK
- Checking the API source code
- Testing API calls
- Asking around
- Forcing platform team to finally type the API :)

Reply:
> Forcing platform team to finally type the API :)

🥺 

Yeah, I'll be.. marking it as possibly undefined for now, and after I go through them all run a pass with the API, see what it returns, even if that'll be pain. Maybe I can read the API source instead and (🤞) gain a better idea..

Reply:
I mean the `name` case is simple. It is a string for named datasets and undefined for unnamed

Reply:
Yep, that one I fixed in [`103c03d` (#193)](https://github.com/apify/apify-client-js/pull/193/commits/103c03d601543d70d3d0b1e52036f07f50a2b016), my remark was more general (since it seems there's even discrepancy between the api docs and tester) 😄